spec/integration/validator_spec.rb

```diff
@@ -114,6 +116,136 @@
     end
   end
 
+  context 'with valid inventory' do
+    let(:inventory) do
+      {
+        'groups' => [
+          {
+            'name' => 'nix',
+            'targets' => [
+              'nix1-example.org',
+              'nix2-example.org'
+            ],
+            'facts' => {
+              'operatingsystem' => 'linux'
+            },
+            'features' => [
+              'puppet-agent'
+            ]
+          },
+          {
+            'name' => 'windows',
+            'targets' => [
+              {
+                'name' => 'win1',
+                'uri' => 'win1-example.org',
+                'transport' => 'winrm'
+              }
+            ],
+            'groups' => [
+              {
+                'name' => 'subgroup',
+                'targets' => [
+                  'win3-example.org',
+                  'win4-example.org'
+                ],
+                'config' => {
+                  'winrm' => {
+                    'ssl' => false
+                  }
+                }
+              }
+            ]
+          }
+        ],
+        'config' => {
+          'transport' => 'ssh',
+          'ssh' => {
+            'host-key-check' => false
```

Comment:
I think it'd be useful to add examples here of keys that are valid for other transports but not `ssh`, and a key that isn't valid for any transport.